### PR TITLE
Add support for multiple hostnames

### PR DESCRIPTION
**Pull request by bdwyertech**
### Description

Add support for adding multiple hostnames.

We use more than one DNS name for our supermarket, primarily for staging purposes.

e.g. supermarket.myorg.net & supermarket.prd.myteam.myorg.net

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by antima-gupta** [comment]
@bdwyertech  chef-style and unit test is failing , please have a look.

**Comment by ayush-billore** [comment]
@bdwyertech Please fix the linting issues as well.

```md
- Error: config/environments/production.rb:119:31: C: [Correctable] Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
- Error: config/environments/production.rb:120:20: C: [Correctable] Layout/SpaceAroundOperators: Operator << should be surrounded by a single space.
- Error: config/environments/production.rb:120:22: C: [Correctable] Layout/ExtraSpacing: Unnecessary spacing detected.
```

**Comment by bdwyertech** [comment]
@ayush-billore @antima-gupta LGTM

File: src/supermarket/config/environments/production.rb [hunk 1]
```diff
@@ -116,6 +116,8 @@
   # The value of ENV["DISABLE_HOST_HEADER_ATTACK"] will be parsed as string.
   # Hence we need to convert string to boolean.
   if ActiveModel::Type::Boolean.new.cast(ENV["DISABLE_HOST_HEADER_ATTACK"]) && ENV["ALLOWED_HOST"].present?
-    config.hosts << ENV["ALLOWED_HOST"]
+    ENV["ALLOWED_HOST"].split(",").each do |host|
+      config.hosts << host
```
Comment (RajeshPaul38) [inline src/supermarket/config/environments/production.rb:120]: Better to use `host.strip` as someone might keep a whitespace between the domain names after the comma.

**Comment by RajeshPaul38** [comment]
We need to change the documentation to adopt this change. @saghoshprogress 
Can't merge as is.

**Comment by dheerajd-msys** [comment]
@bdwyertech Could you please add some doc for this change?

**Comment by RajeshPaul38** [comment]
Hi @bdwyertech 
Thanks for this pull request. We have taken it up in our development pipeline along with required doc changes. Here's the PR that we are working on for your reference: https://github.com/chef/supermarket/pull/2733

**Pull request closed without merging**